File: agent/src/main/java/io/scalac/agent/akka/stream/GraphInterpreterPushAdvice.java
package akka;

import io.scalac.agent.akka.stream.ConnectionOps;
import net.bytebuddy.asm.Advice;

public class GraphInterpreterPushAdvice {

    @Advice.OnMethodEnter
    public static void onPush(
            @Advice.Argument(0) Object currentConnection
    ) {
        ConnectionOps.incrementPushCounter(currentConnection);
    }
}

File: agent/src/main/scala/io/scalac/agent/akka/stream/GraphInterpreterPushAdvice.scala
package io.scalac.agent.akka.stream
import io.scalac.agent.akka.stream.ConnectionOps._
import net.bytebuddy.asm.Advice._

class GraphInterpreterPushAdvice
object GraphInterpreterPushAdvice {

  @OnMethodEnter
  def onPush(@Argument(0) currentConnection: AnyRef): Unit =
    ConnectionOps.incrementPushCounter(currentConnection)
}

class GraphInterpreterPullAdvice
object GraphInterpreterPullAdvice {

  @OnMethodEnter
  def onPull(@Argument(0) currentConnection: AnyRef): Unit =
    incrementPullCounter(currentConnection)
}
